relayer/chain/ethereum/writer.go

```diff
@@ -134,3 +197,118 @@ func (wr *Writer) Write(ctx context.Context, msg *chain.SubstrateOutboundMessage
 
 	return nil
 }
+
+func (wr *Writer) WriteNewSignatureCommitment(ctx context.Context, item *store.BeefyItem, valIndex int) error {
+	beefy, err := item.ToBeefy()
+	if err != nil {
+		wr.log.WithError(err).Error("Error converting database item to Beefy")
+	}
+
+	msg, err := beefy.BuildNewSignatureCommitmentMessage(valIndex)
+	if err != nil {
+		return err
+	}
+
+	// inSet, err := wr.CheckValidatorInSet(ctx, msg.ValidatorPublicKey, msg.ValidatorPosition, msg.ValidatorPublicKeyMerkleProof)
+	// if err != nil {
+	// 	return err
+	// }
+	// if !inSet {
+	// 	return fmt.Errorf("validator address merkle proof failed verification")
+	// }
+
+	contract := wr.polkadotRelayChainBridge
+	if contract == nil {
+		return fmt.Errorf("Unknown contract")
+	}
+
+	blockNumber, err := wr.conn.client.BlockNumber(ctx)
+	if err != nil {
+		return err
+	}
+
+	options := bind.TransactOpts{
+		From:     wr.conn.GetKeyPair().CommonAddress(),
+		Signer:   wr.signerFn,
+		Context:  ctx,
+		GasLimit: 5000000,
+	}
+
+	tx, err := contract.NewSignatureCommitment(&options, msg.Payload,
+		msg.ValidatorClaimsBitfield, msg.ValidatorSignatureCommitment,
+		msg.ValidatorPosition, msg.ValidatorPublicKey, msg.ValidatorPublicKeyMerkleProof)
+	if err != nil {
+		wr.log.WithError(err).Error("Failed to submit transaction")
+		return err
+	}
+
+	wr.log.WithFields(logrus.Fields{
+		"txHash":      tx.Hash().Hex(),
+		"blockNumber": blockNumber,
+	}).Info("New Signature Commitment transaction submitted")
+
+	// Update item's status in database
+	wr.log.Info("2: Updating item status from 'WitnessedCommitment' to 'InitialVerificationTxSent'")
+	instructions := map[string]interface{}{
+		"status":                       store.InitialVerificationTxSent,
+		"initial_verification_tx_hash": tx.Hash(),
+	}
+	updateCmd := store.NewDatabaseCmd(item, store.Update, instructions)
+	wr.beefyMessages <- updateCmd
+
+	return nil
+}
+
+// WriteCompleteSignatureCommitment sends a CompleteSignatureCommitment tx to the PolkadotRelayChainBridge contract
+func (wr *Writer) WriteCompleteSignatureCommitment(ctx context.Context, item *store.BeefyItem) error {
+	beefy, err := item.ToBeefy()
+	if err != nil {
+		wr.log.WithError(err).Error("Error converting database item to Beefy")
+	}
+
+	// Select validator index based on random seed
```

Comment:
seed should come from blockhash of item's completion block, though can worry about getting these parts right in follow up PR once the plumbing/goroutine code is cleaned up and merged

Reply:
Removed in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/ed14d1307e70ccefd10c98516684c226ab613ec2